### PR TITLE
add gh actions

### PR DESCRIPTION
**Pull request by natereid72**
### Description of your changes

Added GH actions to repo

Fixes #

I have:

- [ x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ x] Run `make reviewable` to ensure this PR is ready for review.
- [ x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

GH actions not tested in this PR. Copied and modified actions form similar repo.

i’m still unclear where/how the package version is set with the GH ci. I see some 0.2.0 versions in the file, but GCP plarofrm ref is at 0.0.6 and the ci file in that repo has 0.2.0 too.

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (natereid72) [reply .github/workflows/ci.yaml]: oh, weird, good catch! done.

File: .github/workflows/ci.yaml
Comment (jbw976) [inline .github/workflows/ci.yaml:43]: @natereid72 was wondering how the right version is used to build/push the package.  Note on this line, where we are pushing the configuration to the upbound registry, it's using a version of `steps.tagger.outputs.VERSION_TAG`.

Going up to line 24, the tagger steps computes the version and stores it into a variable called `VERSION_TAG`.  The way it computes the version is by using `git describe`, just like our build submodule does in https://github.com/upbound/build/blob/master/makelib/common.mk#L209-L219, which relies on tags in the repository to generate a version based off that tag.  So once we merge this, we should tag the repo (using the new tag action) as `v0.1.0`, then the CI will pick that tag up in its version calculation and voila!

**Review by jbw976 — APPROVED** [review]
LGTM - I'm not actually sure if we can test this within the PR itself, so we may need to merge it in, run the actions for the first release, and patch anything that doesn't work OK.  I have some high confidence in it though because these work in the platform-ref-aws repo.

**Pull request merged**
